src/auth.rs

```diff
@@ -129,13 +134,11 @@ mod public {
 		}
 
 		#[test]
-		fn test_authenticate_verify_key_too_small() {
+		fn test_authenticate_key_too_small() {
 			let sec_key = SecretKey::generate(31).unwrap();
 			let msg = "what do ya want for nothing?".as_bytes().to_vec();
 
-			let mac_bob = authenticate(&sec_key, &msg).unwrap();
-
-			assert!(authenticate_verify(&mac_bob, &sec_key, &msg).is_err());
+			assert!(authenticate(&sec_key, &msg).is_err());
```

Comment:
After the length check is added in `authenticate_verify()` we want to tests the corresponding error here too.

On top of that, we want to test that the length can actually be in range `32..=64` and that the hazardous `from_slice` error is correctly propagated. So we need to add tests for failure on `65` and pass on `64` as well. Ideally, you could tests this with a `quickcheck!` function. There are a lot of those used in the blake2b hazardous module.

Reply:
Fixed in https://github.com/brycx/orion/pull/88/commits/dbc722e105305e3d5669c0103d676e30d47e7dde